beacon-chain/core/light-client/lightclient.go

```diff
@@ -551,3 +552,204 @@ func NewLightClientUpdateFromOptimisticUpdate(update *ethpbv2.LightClientOptimis
 		SignatureSlot:  update.SignatureSlot,
 	}
 }
+
+func BlockToLightClientHeaderAltair(block interfaces.ReadOnlySignedBeaconBlock) (*ethpbv2.LightClientHeader, error) {
+	parentRoot := block.Block().ParentRoot()
+
+	stateRoot := block.Block().StateRoot()
+
+	bodyRoot, err := block.Block().Body().HashTreeRoot()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get body root")
+	}
+
+	return &ethpbv2.LightClientHeader{
+		Beacon: &ethpbv1.BeaconBlockHeader{
+			Slot:          block.Block().Slot(),
+			ProposerIndex: block.Block().ProposerIndex(),
+			ParentRoot:    parentRoot[:],
+			StateRoot:     stateRoot[:],
+			BodyRoot:      bodyRoot[:],
+		},
+	}, nil
+}
+
+func BlockToLightClientHeaderCapella(ctx context.Context, block interfaces.ReadOnlySignedBeaconBlock) (*ethpbv2.LightClientHeaderCapella, error) {
+	epoch := slots.ToEpoch(block.Block().Slot())
+	if epoch < params.BeaconConfig().CapellaForkEpoch {
+		return nil, fmt.Errorf("creating Capella light client header is not supported before Capella, invalid slot %d", block.Block().Slot())
+	}
```

Comment:
There's a much better way of checking this. You can simply compare `block.Version() != version.Capella`. Same for Deneb and Altair. In fact this check will still allow blocks for forks later than Capella.